cmd/capabilities.go

```diff
@@ -46,40 +47,92 @@ func capsNotDropped(dropped []Capability) (notDropped []Capability, err error) {
 	return
 }
 
+func capAllowed(cap Capability, result *Result) (reason string) {
+	reason = result.Labels["kubeaudit.allow.capability."+strings.ToLower(string(cap))]
+	if strings.ToLower(reason) == "true" {
+		reason = "Unspecified"
+	}
+	return
+}
+
 func checkCapabilities(container Container, result *Result) {
 	if container.SecurityContext == nil {
-		occ := Occurrence{id: ErrorSecurityContextNIL, kind: Error, message: "SecurityContext not set, please set it!"}
+		occ := Occurrence{
+			id:      ErrorSecurityContextNIL,
+			kind:    Error,
+			message: "SecurityContext not set, please set it!",
+		}
 		result.Occurrences = append(result.Occurrences, occ)
 		return
 	}
 
 	if container.SecurityContext.Capabilities == nil {
-		occ := Occurrence{id: ErrorCapabilitiesNIL, kind: Error, message: "Capabilities field not defined!"}
+		occ := Occurrence{
+			id:      ErrorCapabilitiesNIL,
+			kind:    Error,
+			message: "Capabilities field not defined!",
+		}
 		result.Occurrences = append(result.Occurrences, occ)
 		return
 	}
 
 	if container.SecurityContext.Capabilities.Add != nil {
-		result.CapsAdded = container.SecurityContext.Capabilities.Add
-		occ := Occurrence{id: ErrorCapabilitiesAdded, kind: Error, message: "Capabilities were added!"}
-		result.Occurrences = append(result.Occurrences, occ)
+		for _, capAdded := range container.SecurityContext.Capabilities.Add {
+			reasonAllowed := capAllowed(capAdded, result)
+			if reasonAllowed == "" {
+				occ := Occurrence{
+					id:       ErrorCapabilityAdded,
+					kind:     Error,
+					message:  "Capability added",
+					metadata: map[string]string{"CapName": string(capAdded)},
+				}
+				result.Occurrences = append(result.Occurrences, occ)
+			} else {
+				occ := Occurrence{
+					id:      ErrorCapabilityAllowed,
+					kind:    Warn,
+					message: "Capability allowed",
+					metadata: map[string]string{
```

Comment:
metadata could be a separate type Metadata which could have some json serialization associated with it.